### ISSUE TITLE
[Feature_Request] datasets class_to_idx for all datasets

### ISSUE THREAD

**Issue by MrtnMndt** [pytorch/vision#237]
So I noticed that in torchvision.datasets only the ImageFolder class actually has the member called
class_to_idx.

Is there a reason why this isn’t just hard-coded for the other datasets like MNIST, CIFAR and so on, so that one could readily use it for actually getting a class-string for all sorts of visualizations + confusion matrices?

It does seem a little inconvenient to go and manually define the list of all 100 classes for e.g. CIFAR-100.
Do you think this is something that could be added?

**Comment by fmassa** [comment]
Yes, that could be added. Because those are pre-defined datasets, the names of the classes can be hard-coded. Could you send a PR?

**Comment by MrtnMndt** [comment]
K thanks, I will send a PR in the next couple of days after testing every pre-defined dataset first to make sure the ordering isn't somehow messed up in my list, but I am assuming it will all just be in alphabetical order.

**Comment by fmassa** [comment]
The ordering is specified by the dataset, so you should check on each dataset first.
For imagenet for example, [the there is a `syncet` file](https://github.com/szagoruyko/imi-demos/blob/master/imagenet-classification/synset_words.txt) that has the mappings from `n01440764` to `tench, Tinca tinca`. For cifar, the order is described in https://www.cs.toronto.edu/~kriz/cifar.html
